src/libpmemblk/blk.c

```diff
@@ -537,6 +538,14 @@ pmemblk_create(const char *path, size_t bsize, size_t poolsize,
 
 	PMEMblkpool *pbp;
 	pbp = pmemblk_map_common(fd, poolsize, bsize, 0, 1, created);
+	if (pbp && created) {
+		if (chmod(path, mode)) {
```

Comment:
pmemblk_map_common closes the descriptor, so I can't use fchmod here (it was my initial idea too :)

When chmod fails, we close the pool and a couple lines below we unlink this file.

Note: poolset implementation (merged yesterday) completely rewritten this code, so I'll have to also rewrite this patch...